plugins/modules/web_infrastructure/jira.py

```diff
@@ -86,6 +86,26 @@
      - The comment text to add.
      - Note that JIRA may not allow changing field values on specific transitions or states.
 
+  comment_visibility:
+    type: dict
+    required: false
+    description:
+     - Used to specify comment comment visibility.
+     - See U(https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-comments/#api-rest-api-2-issue-issueidorkey-comment-post) for details.
+    options:
+    type:
+      description:
+       - Use type to specify which if the visibility restriction types will be used.
+      type: str
+      required: true
+      choices: [group, role]
+    value:
+      description:
+       - Use value to specify value corresponding to the type of visibility restriction. E.g. name of the group or role.
+      type: str
+      required: true
```

Comment:
That should solve most of the sanity errors :)
```suggestion
      type:
        description:
         - Use type to specify which if the visibility restriction types will be used.
        type: str
        required: true
        choices: [group, role]
      value:
        description:
         - Use value to specify value corresponding to the type of visibility restriction. E.g. name of the group or role.
        type: str
        required: true
```